bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -713,6 +731,45 @@ func (r *KubeadmConfigReconciler) resolveSecretFileContent(ctx context.Context,
 	return data, nil
 }
 
+// resolveUsers maps .Spec.Users into cloudinit.Users, resolving any object references
+// along the way.
+func (r *KubeadmConfigReconciler) resolveUsers(ctx context.Context, cfg *bootstrapv1.KubeadmConfig) ([]bootstrapv1.User, error) {
```

Comment:
Could you kindly add a unit test, something similar to TestKubeadmConfigReconciler_ResolveFiles